### ISSUE TITLE
Endpoint is not callable

### ISSUE THREAD

**Issue by masterofcloud1** [digitalocean/pynetbox#152]
Hello!
I'm using netbox as my db of servers, and performing multiple operations using the data inside of it.
I've tried to use threads to make all the process faster, but then i've encountered that problem:
" 'Endpoint' object is not callable"

The code is going something like that:
from multiprocessing import Pool as ThreadPool
devices = get_devices_from_netbox()
with ThreadPool(10) as pool:
   pool.map(dummy_func,devices)

Looks like something inside of endpoint is not serializable... What do you say?

**Comment by zachmoody** [comment]
Can't really say without knowing what's in `get_devices_from_netbox()`. Can you post its contents?

**Comment by masterofcloud1** [comment]
It is going like that:
`
def get_devices_from_netbox(nb_server,nb_port,token,**filter_args):
  nb_client = pynetbox.api(url=http://{0}:{1).format(nb_server,nb_port), token = token)
  return nb_client.dcim.devices.filter(**filter_args)
`
we get a list of devices which I can iterate on. 
The problem, though, is not with this function - it is in the pool.map

**Comment by zachmoody** [comment]
Had a chance to dig into it a little bit. It looks like it's a result of a change in 4.0 where we started passing the Endpoint object into Response. Will try and find some time to sort it out, but if someone gets it working the the meantime feel free to open a PR.

**Comment by Grokzen** [comment]
@masterofcloud1 Passing data between ThreadPools and Processor pools might go a lot better if you inside your function `get_devices_from_netbox()` manually serialize the data you get back into a normal list/dict data structure and return that instead of the object that pynetbox spits back out when called. My guess is that you are probably seeing is a issue where the pool tries to pickle the objects when it tries to serialize and deserialize them between the threads. Your two solutions is either to make the returned objects work with pickle if that is the serialized format between the pool workers, or you return something that python already can serialize (list + dict) that you then later work with after you have queried all the data you want. I would say that you should sort the data out yourself as that do not require a new release of pynetbox.

**Comment by kartiksubbarao** [comment]
@zachmoody FYI, the patch from @xtelinco worked for me for getting pickle load/dump to work properly with pynetbox. What do you think about merging that code?

**Comment by kartiksubbarao** [comment]
FYI -- I'm currently working around this issue by adding code similar to #168 as a monkey patch at the beginning of the pickle script. Example:

```
#! /usr/bin/python3

import pickle
import pynetbox

def pynetbox_App_init(self, api, name, model=None):
    self.api=api; self.name=name; self._choices=None; self._setmodel()
def pynetbox_App_setmodel(self): self.model=pynetbox_App.models.get(self.name)
def pynetbox_App_getstate(self):
    return { 'api': self.api, 'name': self.name, '_choices': self._choices }
def pynetbox_App_setstate(self, d): self.__dict__.update(d); self._setmodel()
from pynetbox.api import App as pynetbox_App
from pynetbox.models import dcim as pynetbox_dcim, ipam as pynetbox_ipam, \
    virtualization as pynetbox_virtualization, circuits as pynetbox_circuits
pynetbox_App.models = dict(dcim=pynetbox_dcim, ipam=pynetbox_ipam,
    circuits=pynetbox_circuits, virtualization=pynetbox_virtualization)
pynetbox_App.__init__ = pynetbox_App_init
pynetbox_App._setmodel = pynetbox_App_setmodel
pynetbox_App.__getstate__ = pynetbox_App_getstate
pynetbox_App.__setstate__ = pynetbox_App_setstate

netbox = pynetbox.api('http://localhost')
netbox_devices = { device.name:device for device in netbox.dcim.devices.all() }
with open('netbox_devices.pickle', 'wb') as f: pickle.dump(netbox_devices, f)
```

**Comment by zachmoody** [comment]
Should be fixed in #168, closing this out.
